Documentation does not build under sphinx v1.7.7

From travis:

```
10.60s$ cd docs; make html
sphinx-build -b html -W -d _build/doctrees   . _build/html
Running Sphinx v1.7.7
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 55 source files that are out of date
updating environment: 55 added, 0 changed, 0 removed
reading sources... [100%] tutorials/index
Warning, treated as error:
error while formatting arguments for axelrod.strategies.gambler.Gambler: '_Any' object has no attribute '__origin__'
make: *** [html] Error 2
```

In #1201 I've fixed `v1.7.6` in `.travis.yml` but I think that's a temporary fix as readthedocs might update to `v1.7.7`.

--- a/.travis.yml
+++ b/.travis.yml
@@ -13,7 +13,7 @@ before_install:
   - sh -e /etc/init.d/xvfb start
 install:
   - pip install -r requirements.txt
-  - pip install sphinx==1.7.6
+  - pip install sphinx
   - pip install sphinx_rtd_theme
   - pip install docutils==0.12
   - pip install coverage
@@ -21,7 +21,7 @@ install:
   - pip install mypy
 script:
   # Build the documentation
-  - cd docs; make html
+  - cd docs; make clean; make html
   # Run the test suit with coverage
   - cd ..
   - travis_wait 60 coverage run --source=axelrod -m unittest discover

